packages/@n8n/task-runner/src/node-types.ts

```diff
@@ -0,0 +1,63 @@
+import type {
+	IDataObject,
+	INodeType,
+	INodeTypeDescription,
+	INodeTypes,
+	IVersionedNodeType,
+} from 'n8n-workflow';
+
+type VersionedTypes = Map<number, INodeTypeDescription>;
+
+export const DEFAULT_NODETYPE_VERSION = 1;
+
+export class TaskRunnerNodeTypes implements INodeTypes {
+	private versionedNodeTypes: Map<string, VersionedTypes>;
```

Comment:
What's the key here? Version id? Maybe we could add a comment or rename this to `nodeTypesByVersion`